### ISSUE TITLE
RTL8723BE does not show any WiFi networks

### ISSUE THREAD

**Issue by attilaolah** [lwfinger/rtlwifi_new#34]
After installing the current version, I see no networks with an RTL8723BE card. The interface is up, but apparently it does not pick up any broadcast packets.

After that, I tried installing https://github.com/lwfinger/rtl8723be/commit/01b44fd1cab5b1a0f074363c035d4d8be2a7b55f, and that seems to work, I can connect to WiFi networks now.

I do not have the laptop in question with me, but it was an HP ProBook with a fresh copy of Utopic Unicorn (14.10), including the latest updates.

I will bisect this repo next time I have the laptop with me.

**Comment by alzearafat** [comment]
Same problem here. Wifi does not appear in the network manager...  :/

Fedora 21.

**Comment by lwfinger** [comment]
You should switch to http://github.com/lwfinger/rtlwifi_new.git. This repo has not been updates for some time. All the latest are there, not here.

**Comment by attilaolah** [comment]
@lwfinger this **is** https://github.com/lwfinger/rtlwifi_new.

What I was saying is that the module `rtl8723be` from this repo did not work. However, the old one from https://github.com/lwfinger/rtl8723be/commit/01b44fd1cab5b1a0f074363c035d4d8be2a7b55f (with that exact commit) *did* work.

**Comment by lwfinger** [comment]
OK, but what branch are you using? I just built and installed the troy branch and it works perfectly using NetworkManager under openSUSE 13.2.

finger@linux:~/rtlwifi_new> sudo iwlist scan | grep ESSID
                    ESSID:"Ehnen Wifi"
                    ESSID:"radius"
                    ESSID:"HP-Print-A9-Photosmart 5520"
                    ESSID:"lwfdjf-n"
                    ESSID:"ATTUjt6xEa"
                    ESSID:"NETGEAR93"
                    ESSID:"NETGEAR81"
                    ESSID:"J-V Network"
                    ESSID:"HP-Print-0E-Photosmart 6520"
                    ESSID:"RERE"
                    ESSID:"Larry_wpa1"
                    ESSID:"Larry_ac1"
                    ESSID:"KCCoyote,180thSt,E-SE"
                    ESSID:"Chris Stempel's Guest Network"
                    ESSID:"DIRECT-ZX-BRAVIA"

**Comment by attilaolah** [comment]
> what branch are you using?

I tried master. Didn't realised it wasn't stable enough.

Will try the troy branch next time I get my hands on that laptop.

**Comment by lwfinger** [comment]
The master branch works here as well. I have not tried with either Unicorn or Fedora 21, but openSUSE KDE is a charm. Perhaps the NM applet is the problem.

**Comment by attilaolah** [comment]
> Perhaps the NM applet is the problem.

No. `iwlist wlan0 scan` cave me no rscan results most of the time, and sometimes input/output error.

Also using a different module version wouldn't have fixed the nm applet.

**Comment by grdw** [comment]
The same problem exists with `rtl8723AE` driver. Installed the master branch but the wireless networks aren't being detected (`sudo iwlist wlan0 scan` shows no results). I'll try the troy branch to see if that will work.

**Comment by RainbowDashDC** [comment]
Working fine for me on Arch Linux :s using KDE.

**Comment by lwfinger** [comment]
The dmesg output shows that you are using a secure kernel. This code does not generate the secure signature.

**Comment by grdw** [comment]
Allright. Is there anyway to fix that? It sounds like I need to say to my computer "this driver is valid and not a piece of spyware"

**Comment by jamesarbrown** [comment]
Hi, Sorry to jump in here as also having issues with RTL8723be, Ubuntu 15.04 (3.19kernel) HP455G2 all seems ok with system, firmware loaded, no significant errors, but no networks visible.

I have tried many "rmmod rtl8723be" and options like "modprobe fwlps=N ips=N msi=0" but nothing. iwlist scan shows "No Scan Results"

The only discernable issues I can see are -
1) NM reports a "<warn> could not get interface properties: no readable properties for this interface"

2) Whilst modprobe the module the Kernel gives a "AMD-Vi: Event logged [INVALID_DEVICE_REQUEST device=00:00.1 address 0x000000fdf80c0020 flags=0x0a00]

Definitely triggered by the module, but on this machine the device appears to be 00:03.2 if I understand correctly from /sys/devices/pci00000:00/0000:00:03.2/0000:02:00.0/net/wlan0

**Comment by nikolaandj** [comment]
This happens to me on Fedora 21. It used to work for me on OpenSUSE 13.2 when i added the parameter MSI=Y, but since the parameter was deprecated my integrated wifi device is useless! I did have connection drops while I was using OpenSUSE though. The MSI flag was disabled around kernel 3.18 if I remember correctly. Please bring the code for this back since it seemed to work. It doesn't have to be turned on by default, I am just desperate to get my notebook wifi to work since I need it for work. Here are my outputs:

`dmesg | grep 'rtl'` WITH, and WITHOUT `MSI=Y`. Without is 23.* and with is 124.*:

```
[   21.811214] rtlwifi: module verification failed: signature and/or  required key missing - tainting kernel
[   23.324850] Using firmware rtlwifi/rtl8723befw.bin
[   23.337249] ieee80211 phy0: Selected rate control algorithm 'rtl_rc'
[   23.337797] rtlwifi: wireless switch is on
[   23.593505] rtl8723be 0000:02:00.0 wlo1: renamed from wlan0
[  124.434547] rtl8723be: unknown parameter 'msi' ignored
[  124.450948] Using firmware rtlwifi/rtl8723befw.bin
[  124.453385] ieee80211 phy1: Selected rate control algorithm 'rtl_rc'
[  124.453890] rtlwifi: wireless switch is on
[  124.466737] rtl8723be 0000:02:00.0 wlo1: renamed from wlan0
```

And the `cat /proc/version`

```
Linux version 3.19.4-200.fc21.x86_64 (mockbuild@bkernel01.phx2.fedoraproject.org) (gcc version 4.9.2 20150212 (Red Hat 4.9.2-6) (GCC) ) #1 SMP Mon Apr 13 21:43:26 UTC 2015
```

**Comment by lwfinger** [comment]
What disabling of MSI? On kernel 4.1-rc2 from the wireless-drivers-next repo, which should be the same as 3.19.4, a "modinfi rtl8723be" lists the following: Note the underlined entry.

finger@linux:~/rtl8812au> modinfo rtl8723be
filename:       /lib/modules/4.1.0-rc2-wd-next-wl+/kernel/drivers/net/wireless/rtlwifi/rtl8723be/rtl8723be.ko.gz
firmware:       rtlwifi/rtl8723befw.bin
description:    Realtek 8723BE 802.11n PCI wireless
license:        GPL
author:         Realtek WlanFAE <wlanfae@realtek.com>
author:         PageHe  <page_he@realsil.com.cn>
srcversion:     D867B983F049A90211C37BA
alias:          pci:v000010ECd0000B723sv*sd*bc*sc*i*
depends:        rtlwifi,rtl8723-common,rtl_pci,btcoexist,mac80211
intree:         Y
vermagic:       4.1.0-rc2-wd-next-wl+ SMP preempt mod_unload modversions 
parm:           swenc:Set to 1 for software crypto (default 0)
 (bool)
parm:           ips:Set to 0 to not use link power save (default 1)
 (bool)
parm:           swlps:Set to 1 to use SW control power save (default 0)
 (bool)
parm:           fwlps:Set to 1 to use FW control power save (default 1)
 (bool)
parm:           msi:Set to 1 to use MSI interrupts mode (default 0)
==========================================================
parm:           debug:Set debug level (0-5) (default 0) (int)
parm:           disable_watchdog:Set to 1 to disable the watchdog (default 0)
 (bool)
finger@linux:~/rtl8812au> 

MSI interrupts are disabled because that fails on fewer boxes than having them enabled.

Loading the driver with "msi=1" 
finger@linux:~/rtl8812au> sudo modprobe -v rtl8723be msi=1 ips=0
insmod /lib/modules/4.1.0-rc2-wd-next-wl+/kernel/drivers/net/wireless/rtlwifi/rtlwifi.ko.gz 
insmod /lib/modules/4.1.0-rc2-wd-next-wl+/kernel/drivers/net/wireless/rtlwifi/rtl_pci.ko.gz 
insmod /lib/modules/4.1.0-rc2-wd-next-wl+/kernel/drivers/net/wireless/rtlwifi/rtl8723com/rtl8723-common.ko.gz 
insmod /lib/modules/4.1.0-rc2-wd-next-wl+/kernel/drivers/net/wireless/rtlwifi/btcoexist/btcoexist.ko.gz 
insmod /lib/modules/4.1.0-rc2-wd-next-wl+/kernel/drivers/net/wireless/rtlwifi/rtl8723be/rtl8723be.ko.gz ips=0 msi=1 ips=0
finger@linux:~/rtl8812au> 

As to scanning and/or working:

finger@linux:~/rtl8812au> sudo iwlist wlp2s0 scan | grep Cell
          Cell 01 - Address: 00:1A:70:46:BA:B1
          Cell 02 - Address: A4:2B:8C:17:38:66
          Cell 03 - Address: 00:25:9C:3B:AB:A6
          Cell 04 - Address: C0:3F:0E:BE:2B:44
          Cell 05 - Address: 00:14:BF:85:49:FA
          Cell 06 - Address: D8:EB:97:EC:75:8C
          Cell 07 - Address: 14:CF:E2:84:86:30
          Cell 08 - Address: 20:E5:2A:01:F7:EA
          Cell 09 - Address: 74:44:01:47:68:14
finger@linux:~/rtl8812au>

Note: Using the "ips=0" parameter will probably help the stability.

**Comment by jamesarbrown** [comment]
msi=1 ips=0 works for me on Ubuntu 15..... have a feeling I was MSI= before. Thanks. Will see how it goes.

**Comment by lwfinger** [comment]
With Linux, the case is always important.

**Comment by nikolaandj** [comment]
It doesn't work for me with msi=1 ips=0

Here's my full output:

```
[root@localhost nikola]# rmmod rtl8723be
[root@localhost nikola]# modprobe rtl8723be msi=1 ips=0
[root@localhost nikola]# dmesg | grep 'rtl'
[   20.673847] rtlwifi: module verification failed: signature and/or  required key missing - tainting kernel
[   21.881707] Using firmware rtlwifi/rtl8723befw.bin
[   21.895810] ieee80211 phy0: Selected rate control algorithm 'rtl_rc'
[   21.896463] rtlwifi: wireless switch is on
[   22.335602] rtl8723be 0000:02:00.0 wlo1: renamed from wlan0
[   83.787347] Using firmware rtlwifi/rtl8723befw.bin
[   83.789823] ieee80211 phy1: Selected rate control algorithm 'rtl_rc'
[   83.790938] rtlwifi: wireless switch is on
[   83.801313] rtl8723be 0000:02:00.0 wlo1: renamed from wlan0
[  141.908821] Using firmware rtlwifi/rtl8723befw.bin
[  141.910999] ieee80211 phy2: Selected rate control algorithm 'rtl_rc'
[  141.911518] rtlwifi: wireless switch is on
[  141.921331] rtl8723be 0000:02:00.0 wlo1: renamed from wlan0
```

**Comment by lwfinger** [comment]
There was a line missing from the master and kernel_version branches, and the kernel code itself. The rock and troy branches were OK.

I have pushed the fixes for this repo, and a patch will be sent to the kernel maintainer shortly.

When I tested, I must have been using the rock branch, and I missed that line in your log. Sorry.

**Comment by nikolaandj** [comment]
No problem, man. I've compiled the latest version, but still nothing. I've noticed that the msi error message is gone though. Here's the full output:

```
[root@localhost nikola]# rmmod rtl8723be
[root@localhost nikola]# sudo modprobe -v rtl8723be msi=1 ips=0
insmod /lib/modules/3.19.5-200.fc21.x86_64/kernel/drivers/net/wireless/rtlwifi/rtl8723be/rtl8723be.ko.xz msi=1 ips=0
[root@localhost nikola]# dmesg | grep 'rtl'
[   23.341862] rtlwifi: module verification failed: signature and/or  required key missing - tainting kernel
[   24.255086] Using firmware rtlwifi/rtl8723befw.bin
[   24.267946] ieee80211 phy0: Selected rate control algorithm 'rtl_rc'
[   24.268491] rtlwifi: wireless switch is on
[   24.382190] rtl8723be 0000:02:00.0 wlo1: renamed from wlan0
[  175.844051] Using firmware rtlwifi/rtl8723befw.bin
[  175.847097] ieee80211 phy1: Selected rate control algorithm 'rtl_rc'
[  175.847733] rtlwifi: wireless switch is on
[  175.860370] rtl8723be 0000:02:00.0 wlo1: renamed from wlan0
```

**Comment by lwfinger** [comment]
I do not have any really good ideas, but I added some printouts to log when MSI is turned off by an error. Please pull again and try this version.

**Comment by nikolaandj** [comment]
Installed and started with debug level 3. Here's the output:

```
[  385.809923] Using firmware rtlwifi/rtl8723befw.bin
[  385.810093] rtlwifi:_rtl_init_hw_ht_capab():<0-0> 1T1R
[  385.812303] ieee80211 phy5: Selected rate control algorithm 'rtl_rc'
[  385.813524] rtlwifi: wireless switch is on
[  385.813622] rtl_pci:rtl_pci_intr_mode_msi():<0-0> MSI Interrupt Mode!
[  385.822711] rtl8723be 0000:02:00.0 wlo1: renamed from wlan0
[  385.846776] rtl_pci:rtl_pci_start():<0-0>  rtl_pci_start
[  385.846876] btcoexist:rtl_btc_init_hal_vars():<0-0> rtl_btc_init_hal_vars, antNum is 1
[  385.846881] btcoexist:rtl_btc_init_hal_vars():<0-0> rtl_btc_init_hal_vars, bt_exist is 1
[  385.846885] btcoexist:rtl_btc_init_hal_vars():<0-0> rtl_btc_init_hal_vars, bt_type is 8
[  385.852182] rtl8723be:rtl8723be_download_fw():<0-0> normal Firmware SIZE 30746
[  385.852191] rtl8723be:rtl8723be_download_fw():<0-0> Firmware Version(15), Signature(0x5301),Size(32)
[  386.205009] rtl8723be:rtl8723be_enable_hw_security_config():<0-0> PairwiseEncAlgorithm = 0 GroupEncAlgorithm = 0
[  386.205019] rtl8723be:rtl8723be_enable_hw_security_config():<0-0> The SECR-value cc
[  386.261462] rtl_pci:rtl_pci_start():<0-0> rtl_pci_start OK
```

**Comment by nikolaandj** [comment]
Update: Installed 32-bit Fedora, I am able to search for the networks, but the connection is unstable. On 64-bit side, still nothing

**Comment by lwfinger** [comment]
Do you mean you get no output from 'sudo iwlist scan' on 64-bit? The driver works perfectly well for me on 64-bit openSUSE.

**Comment by nikolaandj** [comment]
Yes, I have posted the results earlier in this thread. MSI parameter that worked in earlier kernel versions (up to 3.17, but was broken on 3.18) worked when set to true, you noticed that that parameter didn't work, so you compiled the driver again and it was still broken. On 32-bit it finds networks, without MSI, but is highly unstable. This is on HP Probook 455 G2, and affect several HP notebooks from the Probook line. I can confirm this since some of the people I know do have them and have the exact problem. You can find my outputs in my earlier comments

**Comment by lwfinger** [comment]
I take it that none of my test prints that showed why MSI interrupts were being disabled were logged.

Is it possible to have access to this machine using ssh? If so, I would want to have it running a 64-bit system.

**Comment by nikolaandj** [comment]
I will reinstall 64-bit fedora during the weekend. Can you contact me via email to grant you ssh?

**Comment by lwfinger** [comment]
You can send me the ssh details to Larry.Finger@lwfinger.net.

**Comment by cosmicnag** [comment]
Any solution ? Same problem in HP Pavilion 15 - ab032tx ... no networks
Arch linux kernel 4.2.2-1

**Comment by JHirte** [comment]
I have the same problem with a HP Notebook – 15-af023ng with rtl8723be combined wlan/bt adapter on Gentoo. A scan with iw doesn't show any networks, whereas another laptop with Atheros wlan shows 8-13 different networks. When placing the HP directly to an access-point (about 1-2m), the network is found. iwlist says signal strength is 50/70. Going to the next room, the connection get lost.

I found some workaround for this. 

1. normaly boot the system
2. set the reg domain ('iw reg set DE' for me)
3. wait some time (sleep 4)  and unload the rtl8723be driver (rmmod rtl8723be)
4. wait again (sleep 4) and load rtl8723be again (modprobe rtl8723be)

Now I see some networks here. It's still not as much as the Atheros can see, but at least the same as Windows 7 finds with the same HP notebook. I've tested this with this driver from github as with the one in the kernel sources (latest git). Both work for me.

Drawback: sometimes the reload of the rtl8723be causes a kernel oops in btcoexist. Without the bt-drivers the frequency of the oopes go down, but it still happens. With bt active it happened about 50% of the tests.

**Comment by grdw** [comment]
Problem still maintains with `rtl8723ae`

Tried:

`sudo modprobe rtl8723ae ips=0 debug=3`

Output of `dmesg | grep rtl`

````
[ 1711.992398] rtlwifi:_rtl_init_hw_ht_capab():<0-0> 1T1R
[ 1711.992405] rtlwifi:rtl_regd_init():<0-0> rtl: EEPROM regdomain: 0xb conuntry code: 14
[ 1711.992409] rtlwifi:rtl_regd_init():<0-0> rtl: EEPROM indicates invalid contry codeworld wide 13 should be used
[ 1711.994178] ieee80211 phy3: Selected rate control algorithm 'rtl_rc'
[ 1711.994833] rtlwifi: wireless switch is on
[ 1711.994867] rtl_pci:rtl_pci_intr_mode_legacy():<0-0> Pin-based Interrupt Mode!
[ 1712.033357] rtl_pci:rtl_pci_start():<0-0>  rtl_pci_start
[ 1712.033416] btcoexist:rtl_btc_init_hal_vars():<0-0> rtl_btc_init_hal_vars, antNum is 2
[ 1712.033420] btcoexist:rtl_btc_init_hal_vars():<0-0> rtl_btc_init_hal_vars, bt_exist is 0
[ 1712.033424] btcoexist:rtl_btc_init_hal_vars():<0-0> rtl_btc_init_hal_vars, bt_type is 0
[ 1712.036056] rtl8723ae:rtl8723e_download_fw():<0-0> Firmware Version(21), Signature(0x2302),Size(32)
[ 1712.282122] rtl8723ae:rtl8723e_enable_hw_security_config():<0-0> PairwiseEncAlgorithm = 0 GroupEncAlgorithm = 0
[ 1712.282139] rtl8723ae:rtl8723e_enable_hw_security_config():<0-0> The SECR-value cc
[ 1712.391442] rtl_pci:rtl_pci_start():<0-0> rtl_pci_start OK
[ 1712.398081] rtl8723ae:rtl8723e_dm_bt_coexist():<0-0> [DM][BT], bt_dm_coexist start
[ 1712.398210] rtl8723ae:rtl8723e_dm_bt_bt_hw_counters_monitor():<0-0> High Priority Tx/Rx (reg 0x770)=0(0)/0(0)
[ 1712.398216] rtl8723ae:rtl8723e_dm_bt_bt_hw_counters_monitor():<0-0> Low Priority Tx/Rx (reg 0x774)=0(0)/0(0)
[ 1712.398221] rtl8723ae:rtl8723e_dm_bt_coexist_8723():<0-0> [BTCoex], 2 Ant mechanism
[ 1712.398224] rtl8723ae:_rtl8723e_dm_bt_coexist_2_ant():<0-0> [BTCoex] Get bt info by fw!!
[ 1712.398228] rtl8723ae:rtl8723e_dm_bt_inq_page_monitor():<0-0> [BTCoex], BT Inquiry/page started time : 0x0, cur_time : 0x5630f
[ 1712.398232] rtl8723ae:rtl8723e_dm_bt_is_2_ant_common_action():<0-0> Wifi idle + Bt idle, bt coex mechanism always off!!
[ 1712.398323] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], UPDATE TO NEW COEX SETTING!!
[ 1712.398326] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new bAllOff=0x0/ 0x1
[ 1712.398330] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new b_agc_table_en=0x0/ 0x0
[ 1712.398333] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new b_adc_back_off_on=0x0/ 0x0
[ 1712.398336] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new b2_ant_hid_en=0x0/ 0x0
[ 1712.398340] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new bLowPenaltyRateAdaptive=0x0/ 0x0
[ 1712.398343] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new bRfRxLpfShrink=0x0/ 0x0
[ 1712.398346] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new bRejectAggrePkt=0x0/ 0x0
[ 1712.398350] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new b_tdma_on=0x0/ 0x0
[ 1712.398353] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new tdmaAnt=0x0/ 0x0
[ 1712.398356] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new tdmaNav=0x0/ 0x0
[ 1712.398359] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new tdma_dac_swing=0x0/ 0x0
[ 1712.398363] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new fw_dac_swing_lvl=0x0/ 0x20
[ 1712.398366] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new bTraTdmaOn=0x0/ 0x0
[ 1712.398370] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new traTdmaAnt=0x0/ 0x0
[ 1712.398373] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new traTdmaNav=0x0/ 0x0
[ 1712.398376] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new bPsTdmaOn=0x0/ 0x0
[ 1712.398380] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new psTdmaByte[i]=0x0/ 0x0
[ 1712.398383] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new psTdmaByte[i]=0x0/ 0x0
[ 1712.398387] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new psTdmaByte[i]=0x0/ 0x0
[ 1712.398390] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new psTdmaByte[i]=0x0/ 0x8
[ 1712.398393] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new psTdmaByte[i]=0x0/ 0x0
[ 1712.398397] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new bIgnoreWlanAct=0x0/ 0x0
[ 1712.398400] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new bPtaOn=0x0/ 0x0
[ 1712.398404] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new val_0x6c0=0x0/ 0x5a5aaaaa
[ 1712.398407] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new val_0x6c8=0x0/ 0xcc
[ 1712.398410] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new val_0x6cc=0x0/ 0x3
[ 1712.398413] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new sw_dac_swing_on=0x0/ 0x0
[ 1712.398417] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new sw_dac_swing_lvl=0x0/ 0xc0
[ 1712.398420] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new wlanActHi=0x0/ 0x10
[ 1712.398423] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new wlanActLo=0x0/ 0x10
[ 1712.398427] rtl8723ae:rtl8723e_dm_bt_set_bt_dm():<0-0> [BTCoex], original/new btRetryIndex=0x0/ 0x2
[ 1712.404423] rtl8723ae:_rtl8723e_dm_bt_coexist_2_ant():<0-0> Action 2-Ant common.
[ 1712.404427] rtl8723ae:rtl8723e_dm_bt_is_same_coexist_state():<0-0> [DM][BT], Coexist state changed!!
[ 1712.404431] rtl8723ae:rtl8723e_dm_bt_coexist_8723():<0-0> [BTCoex], Coexist State[bitMap] change from 0x0       0 to 0x040000102
[ 1714.393985] rtl8723ae:rtl8723e_dm_bt_coexist():<0-0> [DM][BT], bt_dm_coexist start
[ 1714.394105] rtl8723ae:rtl8723e_dm_bt_bt_hw_counters_monitor():<0-0> High Priority Tx/Rx (reg 0x770)=0(0)/2(2)
[ 1714.394117] rtl8723ae:rtl8723e_dm_bt_bt_hw_counters_monitor():<0-0> Low Priority Tx/Rx (reg 0x774)=0(0)/0(0)
[ 1714.394125] rtl8723ae:rtl8723e_dm_bt_coexist_8723():<0-0> [BTCoex], 2 Ant mechanism
[ 1714.394132] rtl8723ae:_rtl8723e_dm_bt_coexist_2_ant():<0-0> [BTCoex] Get bt info by fw!!
[ 1714.394140] rtl8723ae:rtl8723e_dm_bt_inq_page_monitor():<0-0> [BTCoex], BT Inquiry/page started time : 0x0, cur_time : 0x56503
[ 1714.394147] rtl8723ae:rtl8723e_dm_bt_is_2_ant_common_action():<0-0> Wifi idle + Bt idle, bt coex mechanism always off!!
[ 1714.394153] rtl8723ae:_rtl8723e_dm_bt_coexist_2_ant():<0-0> Action 2-Ant common.
[ 1714.394159] rtl8723ae:rtl8723e_dm_bt_is_same_coexist_state():<0-0> [DM][BT], Coexist state do not chang!!
[ 1716.397897] rtl8723ae:rtl8723e_dm_bt_coexist():<0-0> [DM][BT], bt_dm_coexist start
[ 1716.398026] rtl8723ae:rtl8723e_dm_bt_bt_hw_counters_monitor():<0-0> High Priority Tx/Rx (reg 0x770)=0(0)/1(1)
[ 1716.398039] rtl8723ae:rtl8723e_dm_bt_bt_hw_counters_monitor():<0-0> Low Priority Tx/Rx (reg 0x774)=0(0)/0(0)
[ 1716.398047] rtl8723ae:rtl8723e_dm_bt_coexist_8723():<0-0> [BTCoex], 2 Ant mechanism
[ 1716.398054] rtl8723ae:_rtl8723e_dm_bt_coexist_2_ant():<0-0> [BTCoex] Get bt info by fw!!
[ 1716.398063] rtl8723ae:rtl8723e_dm_bt_inq_page_monitor():<0-0> [BTCoex], BT Inquiry/page started time : 0x0, cur_time : 0x566f8
[ 1716.398070] rtl8723ae:rtl8723e_dm_bt_is_2_ant_common_action():<0-0> Wifi idle + Bt idle, bt coex mechanism always off!!
[ 1716.398077] rtl8723ae:_rtl8723e_dm_bt_coexist_2_ant():<0-0> Action 2-Ant common.
[ 1716.398083] rtl8723ae:rtl8723e_dm_bt_is_same_coexist_state():<0-0> [DM][BT], Coexist state do not chang!!
[ 1718.402081] rtl8723ae:rtl8723e_dm_bt_coexist():<0-0> [DM][BT], bt_dm_coexist start
[ 1718.402212] rtl8723ae:rtl8723e_dm_bt_bt_hw_counters_monitor():<0-0> High Priority Tx/Rx (reg 0x770)=0(0)/2(2)
[ 1718.402221] rtl8723ae:rtl8723e_dm_bt_bt_hw_counters_monitor():<0-0> Low Priority Tx/Rx (reg 0x774)=0(0)/0(0)
[ 1718.402228] rtl8723ae:rtl8723e_dm_bt_coexist_8723():<0-0> [BTCoex], 2 Ant mechanism
[ 1718.402233] rtl8723ae:_rtl8723e_dm_bt_coexist_2_ant():<0-0> [BTCoex] Get bt info by fw!!
[ 1718.402240] rtl8723ae:rtl8723e_dm_bt_inq_page_monitor():<0-0> [BTCoex], BT Inquiry/page started time : 0x0, cur_time : 0x568ed
[ 1718.402246] rtl8723ae:rtl8723e_dm_bt_is_2_ant_common_action():<0-0> Wifi idle + Bt idle, bt coex mechanism always off!!
[ 1718.402251] rtl8723ae:_rtl8723e_dm_bt_coexist_2_ant():<0-0> Action 2-Ant common.
[ 1718.402256] rtl8723ae:rtl8723e_dm_bt_is_same_coexist_state():<0-0> [DM][BT], Coexist state do not chang!!
[ 1720.406154] rtl8723ae:rtl8723e_dm_bt_coexist():<0-0> [DM][BT], bt_dm_coexist start
[ 1720.406282] rtl8723ae:rtl8723e_dm_bt_bt_hw_counters_monitor():<0-0> High Priority Tx/Rx (reg 0x770)=0(0)/1(1)
[ 1720.406294] rtl8723ae:rtl8723e_dm_bt_bt_hw_counters_monitor():<0-0> Low Priority Tx/Rx (reg 0x774)=0(0)/0(0)
[ 1720.406302] rtl8723ae:rtl8723e_dm_bt_coexist_8723():<0-0> [BTCoex], 2 Ant mechanism
[ 1720.406309] rtl8723ae:_rtl8723e_dm_bt_coexist_2_ant():<0-0> [BTCoex] Get bt info by fw!!
[ 1720.406317] rtl8723ae:rtl8723e_dm_bt_inq_page_monitor():<0-0> [BTCoex], BT Inquiry/page started time : 0x0, cur_time : 0x56ae2
[ 1720.406324] rtl8723ae:rtl8723e_dm_bt_is_2_ant_common_action():<0-0> Wifi idle + Bt idle, bt coex mechanism always off!!
[ 1720.406330] rtl8723ae:_rtl8723e_dm_bt_coexist_2_ant():<0-0> Action 2-Ant common.
[ 1720.406337] rtl8723ae:rtl8723e_dm_bt_is_same_coexist_state():<0-0> [DM][BT], Coexist state do not chang!!
[ 1722.410592] rtl8723ae:rtl8723e_dm_bt_coexist():<0-0> [DM][BT], bt_dm_coexist start
[ 1722.410718] rtl8723ae:rtl8723e_dm_bt_bt_hw_counters_monitor():<0-0> High Priority Tx/Rx (reg 0x770)=0(0)/2(2)
[ 1722.410729] rtl8723ae:rtl8723e_dm_bt_bt_hw_counters_monitor():<0-0> Low Priority Tx/Rx (reg 0x774)=0(0)/0(0)
[ 1722.410737] rtl8723ae:rtl8723e_dm_bt_coexist_8723():<0-0> [BTCoex], 2 Ant mechanism
[ 1722.410744] rtl8723ae:_rtl8723e_dm_bt_coexist_2_ant():<0-0> [BTCoex] Get bt info by fw!!
[ 1722.410752] rtl8723ae:rtl8723e_dm_bt_inq_page_monitor():<0-0> [BTCoex], BT Inquiry/page started time : 0x0, cur_time : 0x56cd7
[ 1722.410759] rtl8723ae:rtl8723e_dm_bt_is_2_ant_common_action():<0-0> Wifi idle + Bt idle, bt coex mechanism always off!!
[ 1722.410765] rtl8723ae:_rtl8723e_dm_bt_coexist_2_ant():<0-0> Action 2-Ant common.
[ 1722.410771] rtl8723ae:rtl8723e_dm_bt_is_same_coexist_state():<0-0> [DM][BT], Coexist state do not chang!!
[ 1724.414532] rtl8723ae:rtl8723e_dm_bt_coexist():<0-0> [DM][BT], bt_dm_coexist start
[ 1724.414659] rtl8723ae:rtl8723e_dm_bt_bt_hw_counters_monitor():<0-0> High Priority Tx/Rx (reg 0x770)=0(0)/2(2)
[ 1724.414665] rtl8723ae:rtl8723e_dm_bt_bt_hw_counters_monitor():<0-0> Low Priority Tx/Rx (reg 0x774)=0(0)/0(0)
[ 1724.414670] rtl8723ae:rtl8723e_dm_bt_coexist_8723():<0-0> [BTCoex], 2 Ant mechanism
[ 1724.414673] rtl8723ae:_rtl8723e_dm_bt_coexist_2_ant():<0-0> [BTCoex] Get bt info by fw!!
[ 1724.414678] rtl8723ae:rtl8723e_dm_bt_inq_page_monitor():<0-0> [BTCoex], BT Inquiry/page started time : 0x0, cur_time : 0x56ecc
[ 1724.414682] rtl8723ae:rtl8723e_dm_bt_is_2_ant_common_action():<0-0> Wifi idle + Bt idle, bt coex mechanism always off!!
[ 1724.414686] rtl8723ae:_rtl8723e_dm_bt_coexist_2_ant():<0-0> Action 2-Ant common.
[ 1724.414689] rtl8723ae:rtl8723e_dm_bt_is_same_coexist_state():<0-0> [DM][BT], Coexist state do not chang!!
````
I can't make any sense of the above though. Maybe someone can .

````
sudo iwlist wlan0 scan

wlan0     No scan results
````

**Comment by lwfinger** [comment]
Rather than throw random driver options at the problem, let's use a few basic debugging concepts:
 (A) All of the power-setting params (ips, swlps, fwlps) control power-save operations for a connection. As you have not gotten such a connection, they will make no difference here.
(B) There shouldseldom be a reason to use software encoding (swenc) in the field. It is only there to be able to test new firmware, or possibly to salvage defective hardware. It will only be in effect after a connection is made, thus we can ignore it for now.
(C) Option msi controls whether the interface uses standard interrupts or Message Signalled Interrupts. Most computers will work with either, but a few need MSI. The default is to use conventional interrupts. To switch, load the module with the "msi=1" option. I did discover that the MSI option code was not being initialized correctly, thus you should do a 'git pull' before any further trials.

**Comment by grdw** [comment]
> Rather than throw random driver options at the problem,

Consider it a "I really want my wifi card to just function properly, let's just try all of it I'm desperate" move. And let's just say I tried a lot to fix it and it's been a big pain in the ass. 

> thus you should do a 'git pull' before any further trials.

I do want to point out I use the `rtl8723ae` instead of the `rtl8723be` but the issue is the same.

487c5ee clearly might fix thing for the `be` but I can guess that it doesn't for the `ae`.

**Comment by lwfinger** [comment]
If you have an RTL8723AE, then why are you posting under an RTL8723BE title? What am I to suppose?

**Comment by mmarquezs** [comment]
I have what it seems to be the same problem in a HP laptop 15-ac102ns with the rtl8723be. Even tought after some testing it seems it actually detects networks but only if the networks are really really near, I mean my own wireless in my home it doesn't get detected and its really near,... but if I make a hotspot with my phone just next to the laptop it show up. I tried msi=1 and other parameters and either it didn't work or I ended restarting because btcoexist crashed. (btw there is any way to have the btcoexist disabled so i can test thing without having to reboot due to it? I don't need bluetooth right now) . One single time it seems it worked for some strange reason... but i haven't been able to recreate it :( Even though it worked for some seconds like it showed my wifi and then sometimes nothing and then sometimes it showed up... ... 

On windows I get 15 wireless access points,... so clearly something is messed up :S 

I use ArchLinux with kernel 4.2.5 and this drivers installed from here (master branch), i don't know what logs to provide or what it would be useful.

**Comment by nikolaandj** [comment]
Try leaving `msi=1` in the `.conf` file. Also try turning off any power saving options. It seems that they lower your scan range.

**Comment by mmarquezs** [comment]
Well I have the file "/etc/modprobe.d/rt8723be" as this:
>options rtl8723be msi=1 fwlps=N ips=N

But the same results :( .
BUT I found a FIX.. it works... but it feels bad...  :P It's the weirdest fix i have seen in my life. I don't have a clue why it works... and i did find it testing things. but it works every single time... 

Since the drivers didn't seems to work I wanted to try ndiswrapper... sadly the drivers that I found were for windows 7 o windows 8 but not windows xp . I tried and it seems it installed them but.. when starting ndiswrapper no interface was available and it said it found errors loading that coudn't load the driver . So no interface to bring up or use to scan...but then trying things like loading this drivers etc.. sudennly it worked and well I found that doing this it makes it work... but no clue why and clearly it's an horrible fix... and obviusly this doesn't fix the driver it's just a work around for me to be avaible to use the wifi. 

The only I can think is that when ndiswrapper is loading the driver although it fails,during the loading it  makes something that makes it work... but i have no idea. 

>rmmod ndiswrapper
>ip link set wlp13s0 up
>rmmod rtl8723be
>modprobe -v ndiswrapper
>modprobe rtl8723be
>sleep 2
>ip link set wlp13s0 up
>rmmod ndiswrapper

**Comment by lwfinger** [comment]
As you probably know, I do not see this problem. One of the main differences is that the only versions of Windows that I run are contained in virtual machines, thus they never touch the real PCI hardware, which includes the wifi device. Windows is known to leave hardware in a crazy state. In my setup, power is always removed from the Realtek devices on every reboot.

When you boot from Windows to Linux, do you perform a cold reboot? If so, and you leave the machine turned off for at least 15 sec, then the RTL8723BE hardware should be cleared of any settings added by the Windows driver. Please test that without your "fix". If that does not help, then disconnect the power and remove the battery while shut down.

I have no idea what your "fix" is doing. According to what you said, only ndiswrapper is loading, thus nothing should be happening with the device. Unfortunately, the debugging of I/O on a PCI device requires special kernel options, and is difficult to set up. I'm not sure that I could work you through it.

**Comment by mmarquezs** [comment]
Except the first boots, the rest was me trying to fix this so rebooting from Linux to Linux. And it was reboots, no cold boots. 

I tried a cold boot, first I did remove ndiwswrapper form the system plus my "fix" and the I did the cold boot.But I get some strange behaviour... now it shows my home access point when I scan... but that's the only one, when using my fix it showed all the others, and well it's the nearest one so I guess the range is still ,for  some reason, not what it should be, also I tried to connect to it but it didn't work. 

And about ndiswrapper yes unless it's lying to me is just the wrapper that's loading but nothing else, but who knows what happens under the hood....because if i remove it it stops working, so something is doing... This is the messages I had that made me thing that it doesn't load the driver:  

> [   13.827562] ndiswrapper (load_sys_files:200): couldn't prepare driver 'netrtwlane'
> [   13.828285] ndiswrapper (load_wrap_driver:103): couldn't load driver netrtwlane; check system log for messages from 'loadndisdriver'

**Comment by lwfinger** [comment]
It is possible that your script is somehow setting the antenna. I have attempted adding a module option to select which antenna will be used. The new code is in the "test" branch of the repo and is triggered by using the "ant_sel=Y" module option for rtl8723be. Y is either 1 or 2.

**Comment by alfbr** [comment]
This issue bit me on a HP Pavilion laptop, with rtl8723be. I am trying to get this working on Debian Jessie/stable, which has the 3.16 kernel.

iwlist scan yelds "No scan results" by default. I have tested this with the distro driver, in addition to master and troy branch of rtlwifi_new. Loading the driver with the option msi=1 will have scanning working on all three versions of the driver, but none of them is able to connect. Any hints is greatly appreciated.

**Comment by lwfinger** [comment]
If it sees networks while scanning, then any connection problems are likely in some other part of the user code. What distro are you using, and what controls the network?

**Comment by jamesarbrown** [comment]
Mine hp455g2 pro book /etc/modprobe.d/rtl8723be.conf options‎ rtl8723be msi=1 ips=1But I note that occasionally and with heavy traffic it breaks. So I have a script to rmmod (Inc btcoexist) and modprobe it back on‎.... even that can fully freeze machine if done in wrong order                                                                                                                                                                                                                                                                                                                                        Sent from my BlackBerry 10 smartphone on the O2 network.                                                                                                                                                                                                                From: lwfingerSent: Thursday, 31 December 2015 01:37To: lwfinger/rtlwifi_newReply To: lwfinger/rtlwifi_newCc: James A R BrownSubject: Re: [rtlwifi_new] RTL8723BE does not show any WiFi networks (#34)If it sees networks while scanning, then any connection problems are likely in some other part of the user code. What distro are you using, and what controls the network?

—Reply to this email directly or view it on GitHub.

**Comment by mmarquezs** [comment]
> It is possible that your script is somehow setting the antenna. I have attempted adding a module option to select which antenna will be used. The new code is in the "test" branch of the repo and is triggered by using the "ant_sel=Y" module option for rtl8723be. Y is either 1 or 2.

I have tested the test branch and nothing. Same results. I have seen this other issue https://github.com/lwfinger/rtlwifi_new/issues/80 and he tried to use this options but like me it didn't work. But he changed the antenna physically and it worked,so  it could be that the code is somehow wrong and it keeps using the antenna 1?

**Comment by alfbr** [comment]
Thanks for the swift reply @lwfinger . As mentioned I am running the latest Debian stable, i.e.,  8/Jessie. I am testing from the KDE desktop, so connection with plasma-nm version 0.9.3.4, which again uses networkmanager version 0.9.10.

I tried setting up wireless using iwconfig directly without success (disabling networkmanager). Also tried the Debian way setting up wireless in /etc/network/interfaces (any interfaces set-up there will be ignored by networkmanager). Neither resulted in a working connection.

I am running openwrt on the router, and set-up an unsecure interface to rule out any authentication issues, so no need for wep/wpa/wpa2.

I confirmed with a second laptop here with identical software set-up (a thinkpad with intel wireless running KDE spin of Debian 8) to ensure that all approaches resulted in working network connection.

Should I open a separate issue for this?

**Comment by lwfinger** [comment]
@mmarquezs: I said earlier that the code was untested. In fact, I cannot duplicate the problem, thus no way to test. Too bad your vendor cannot burn the efuse correctly.

@alfbr: Yes, that should be a separate issue.

**Comment by lwfinger** [comment]
Realtek just informed me that the routine to select the connected antenna automatically has been delayed, but they supplied a corrected routine to fix the "ant_sel" module parameter.

To access this new code, use the "rock.new_btcoex" branch and load the module with the "ant_sel=Y" option, where Y is 1 or 2.

**Comment by mmarquezs** [comment]
Just tested it. Sadly same results :( 

Just in case I did something wrong, I did this:
I removed all my old fixes and /etc/modprobe.d/rtl8723be.conf . Then I downloaded the rock.new_btcoex branch, and then:
$ make
$ make install
Then I did a shutdown without power cable / battery so it's a cold boot.
Then I booted and I did 
$ sudo modprove -rv rtl8723be
$ sudo modprove -v rtl8723be ant_sel=1
$ sudo ip link set wlp13s0 up
$ sudo iw dev wlp13s0 scan
Nothing, I did set up my mobile phone as an AP near the laptop and then i did it again and it shows only this one. 
Then again:
$ sudo modprove -rv rtl8723be
Now changing antenna
$ sudo modprove -v rtl8723be ant_sel=2
$ sudo ip link set wlp13s0 up
$ sudo iw dev wlp13s0 scan
Nothing, I did set up my mobile phone as an AP near the laptop and then i did it again and it shows only this one. 

It both cases i get the same results sadly. I'll wait for other people to test to see if I'm the only one or if it works for somebody.

**Comment by lwfinger** [comment]
Assuming you meant modprobe rather than modprove, then it seems that the code does not work.

Was there any difference in the signal strength?

**Comment by lwfinger** [comment]
I found the bug and just pushed code that works here. With a single antenna connected to connector 2 and sudo modprobe -v rtl8723be ant_sel=2, I get

finger@linux:~/rtlwifi_new> sudo iwlist wlp2s0 scan | egrep "SSID|level"
Quality=70/70 Signal level=-38 dBm
ESSID:"Larry_ac"

With sudo modprobe -v rtl8723be ant_sel=1, I get

finger@linux:~/rtlwifi_new> sudo iwlist wlp2s0 scan | egrep "SSID|level"
Quality=30/70 Signal level=-80 dBm
ESSID:"lwfdjf-n"
Quality=44/70 Signal level=-66 dBm
ESSID:"NETGEAR81"
Quality=24/70 Signal level=-86 dBm
ESSID:"radius"

**Comment by mmarquezs** [comment]
Thanks a lot, it works fine now :D

**Comment by lwta08** [comment]
Thanks a lot. Solved my problems on an HP x360 13-s.

**Comment by Rajat-Goyal** [comment]
"I found the bug and just pushed code that works here."

"here" means which place ? could you just help me exactly locate the point which i have to clone and then make install etc ?

**Comment by lwta08** [comment]
... I got from here:
https://github.com/lwfinger/rtlwifi_new/tree/rock.new_btcoex

**Comment by lwfinger** [comment]
That is the source with the fix. You need to unpack that source, "make", and "sudo make install". At that point use your favorite editor to create (as root) a file /etc/modprobe.d/50-rtl8723be.conf containing a single line with "options rtl8723be ips=0 ant_sel=2". Then reboot. Your wifi should now work.

**Comment by Rajat-Goyal** [comment]
do i need to clone the master branch or the rock.new_btcoex branch ??

**Comment by lwfinger** [comment]
That error is because you have not installed the kernel header package. I do not know what ARCH calls it, thus you will need to do some searching, or ask on their forum what is needed to build out-of-kernel modules.

You have the correct rtlwifi_new branch. Please be aware that because you downloaded the zip file from GitHub.com, it will be necessary to redownload everything when I change any part of it. Those people that use git to clone the repo only need to download the changed part,

**Comment by Rajat-Goyal** [comment]
you mean to say using something like 
git clone -b rock.new_btcoex https://github.com/lwfinger/rtlwifi_new.git
will be better instead of a zip file ?

**Comment by lwfinger** [comment]
Yes. If I change a single character in the sources, you need to download the entire source when you use a zip file. With git, a simple "git pull" will download only the part that changed.

**Comment by Rajat-Goyal** [comment]
thanks a lot !! it works now. It detects all the wifi connection in the range quite well now :D

**Comment by Rajat-Goyal** [comment]
ok the wifi seems to detect all networks properly. But there is another problem which i tried googling but haven't found any working solutions yet. Each time i suspend my laptop and then get it back on . the wifi appears as disabled by hardware. There is no hardware switch and i am not able to use anything to get it back on. I have to reboot my system each time to work around this problem.

**Comment by lwfinger** [comment]
After your laptop is suspended, please run the command "rfkill list all" and post the results.

**Comment by Rajat-Goyal** [comment]
results of "rfkill list all"
```
[shrey@rajat ~]$ rfkill list all
0: phy0: Wireless LAN
	Soft blocked: no
	Hard blocked: yes
```

**Comment by lwfinger** [comment]
You do have a hardware switch. That is what is causing the hard block above. Either one would kill the wireless. If a soft block, rfkill could be used to clear it, but not a hard block.

You need to check what wmi routine is being loaded. "lsmod | grep wmi" should show that. If blank, please post the entire output of lsmod.

**Comment by Rajat-Goyal** [comment]
i cant find a hardware switch as any key on my laptop :/ but yeah it seems there is something . Here is the out put 
```
[shrey@rajat cf]$ lsmod | grep wmi 
mxm_wmi                16384  1 nouveau
hp_wmi                 16384  0
sparse_keymap          16384  1 hp_wmi
rfkill                 20480  5 cfg80211,hp_wmi,bluetooth
wmi                    20480  3 hp_wmi,mxm_wmi,nouveau
```

**Comment by lwfinger** [comment]
For an HP laptop, hp_wmi is the driver that interprets the various FN keys. From Google searching, it seems that FN-F12 is the one that sets airplane mode; however, even the Windows users have a lot of trouble restoring wireless after a crash, No one mentioned suspending the machine. I do not have an HP so I have no idea what problems there might be with that module.

One thing to try: Run "sudo modprobe -rv hp_wmi" to unload that driver after a fresh reboot. Then try suspending and restoring to see what happens. None of the FN-F keys will work with that driver removed, but if unloading it before suspend works, then we can try an unload before suspend and a reload after resuming.

**Comment by Rajat-Goyal** [comment]
hey,
i tried "sudo modprobe -rv hp_wmi" after a fresh reboot. But this didn't work. The wifi still remains disabled by hardware. Fn + F12 is for flight mode and that seems enabled after suspension. 
While trying "sudo modprobe -rv hp_wmi" i tried using the Fn keys after removing the module. My brightness and volume buttons are all on Fn keys and everythin still works even after removing the hp_wmi module 
just incase if you might need here is the output for "lsmod" 

```
[shrey@rajat ~]$ lsmod
Module                  Size  Used by
fuse                   90112  5
sha256_ssse3           28672  1
sha256_generic         24576  1 sha256_ssse3
hmac                   16384  1
drbg                   32768  1
ansi_cprng             16384  0
ctr                    16384  1
ccm                    20480  1
intel_rapl             20480  0
snd_hda_codec_hdmi     49152  1
x86_pkg_temp_thermal    16384  0
intel_powerclamp       16384  0
coretemp               16384  0
kvm_intel             163840  0
kvm                   450560  1 kvm_intel
crct10dif_pclmul       16384  0
crc32_pclmul           16384  0
mousedev               20480  0
crc32c_intel           24576  0
aesni_intel           167936  2
joydev                 20480  0
snd_hda_codec_realtek    69632  1
snd_hda_codec_generic    69632  1 snd_hda_codec_realtek
aes_x86_64             20480  1 aesni_intel
arc4                   16384  2
rtl8723be             135168  0
btcoexist             409600  1 rtl8723be
uvcvideo               86016  0
lrw                    16384  1 aesni_intel
gf128mul               16384  1 lrw
hp_wmi                 16384  0
ppdev                  20480  0
glue_helper            16384  1 aesni_intel
r8169                  77824  0
sparse_keymap          16384  1 hp_wmi
iTCO_wdt               16384  0
iTCO_vendor_support    16384  1 iTCO_wdt
rtl_pci                40960  1 rtl8723be
rtlwifi               118784  3 btcoexist,rtl_pci,rtl8723be
btusb                  40960  0
btrtl                  16384  1 btusb
videobuf2_vmalloc      16384  1 uvcvideo
btbcm                  16384  1 btusb
videobuf2_memops       16384  1 videobuf2_vmalloc
btintel                16384  1 btusb
mac80211              647168  3 rtl_pci,rtlwifi,rtl8723be
videobuf2_core         40960  1 uvcvideo
v4l2_common            16384  1 videobuf2_core
videodev              143360  3 uvcvideo,v4l2_common,videobuf2_core
snd_hda_intel          32768  10
media                  20480  2 uvcvideo,videodev
ablk_helper            16384  1 aesni_intel
nouveau              1441792  0
mii                    16384  1 r8169
bluetooth             450560  5 btbcm,btrtl,btusb,btintel
cfg80211              466944  2 mac80211,rtlwifi
snd_hda_codec         106496  4 snd_hda_codec_realtek,snd_hda_codec_hdmi,snd_hda_codec_generic,snd_hda_intel
rfkill                 20480  6 cfg80211,hp_wmi,bluetooth
snd_hda_core           49152  5 snd_hda_codec_realtek,snd_hda_codec_hdmi,snd_hda_codec_generic,snd_hda_codec,snd_hda_intel
snd_hwdep              16384  1 snd_hda_codec
snd_pcm                86016  4 snd_hda_codec_hdmi,snd_hda_codec,snd_hda_intel,snd_hda_core
i915                 1081344  25
mxm_wmi                16384  1 nouveau
evdev                  24576  16
ttm                    77824  1 nouveau
input_leds             16384  0
cryptd                 20480  2 aesni_intel,ablk_helper
psmouse               118784  0
mac_hid                16384  0
pcspkr                 16384  0
drm_kms_helper        102400  2 i915,nouveau
snd_timer              28672  1 snd_pcm
serio_raw              16384  0
drm                   286720  11 ttm,i915,drm_kms_helper,nouveau
snd                    65536  28 snd_hda_codec_realtek,snd_hwdep,snd_timer,snd_hda_codec_hdmi,snd_pcm,snd_hda_codec_generic,snd_hda_codec,snd_hda_intel
thermal                20480  0
hp_accel               28672  0
battery                20480  0
led_class              16384  2 hp_accel,input_leds
parport_pc             28672  0
hp_wireless            16384  0
wmi                    20480  3 hp_wmi,mxm_wmi,nouveau
parport                40960  2 ppdev,parport_pc
lis3lv02d              20480  1 hp_accel
processor_thermal_device    16384  0
int340x_thermal_zone    16384  1 processor_thermal_device
fjes                   28672  0
intel_soc_dts_iosf     16384  1 processor_thermal_device
mei_me                 32768  0
mei                    81920  1 mei_me
intel_gtt              20480  1 i915
syscopyarea            16384  1 drm_kms_helper
sysfillrect            16384  1 drm_kms_helper
sysimgblt              16384  1 drm_kms_helper
fb_sys_fops            16384  1 drm_kms_helper
i2c_algo_bit           16384  2 i915,nouveau
int3400_thermal        16384  0
acpi_thermal_rel       16384  1 int3400_thermal
input_polldev          16384  1 lis3lv02d
soundcore              16384  1 snd
shpchp                 32768  0
video                  32768  2 i915,nouveau
ac                     16384  0
acpi_pad               20480  0
lpc_ich                24576  0
iosf_mbi               16384  2 intel_soc_dts_iosf,intel_rapl
i2c_i801               20480  0
button                 16384  2 i915,nouveau
processor              32768  0
sch_fq_codel           20480  6
ip_tables              28672  0
x_tables               28672  1 ip_tables
ext4                  507904  2
crc16                  16384  2 ext4,bluetooth
mbcache                20480  1 ext4
jbd2                   90112  1 ext4
sr_mod                 24576  0
sd_mod                 36864  3
cdrom                  49152  1 sr_mod
atkbd                  24576  0
libps2                 16384  2 atkbd,psmouse
ahci                   36864  2
libahci                28672  1 ahci
libata                196608  2 ahci,libahci
xhci_pci               16384  0
xhci_hcd              155648  1 xhci_pci
scsi_mod              151552  3 libata,sd_mod,sr_mod
usbcore               196608  4 btusb,uvcvideo,xhci_hcd,xhci_pci
usb_common             16384  1 usbcore
i8042                  24576  2 hp_accel,libps2
serio                  20480  7 serio_raw,atkbd,i8042,hp_accel,psmouse
```

**Comment by lwfinger** [comment]
The "flight mode" operation is exactly what is controlling the rfkill signal. Are you sure that toggling that one does not restore wifi after suspension?

**Comment by Rajat-Goyal** [comment]
i cannot toggle the "flight mode" operation after i restore from suspension. And yes i tried the remove module thing thrice.. it made no differece. Every Fn key works even after removing the hp_wmi module

**Comment by lwfinger** [comment]
It seems to me that the problem is in whatever driver is controlling the Fn keys. I'm not sure what sparse_keymap does. Driver mxm_wmi likely handles the brightness, but you will not be able to remove it.

Does Fn F12 control the radio if you have not suspended, or does it always fail.

**Comment by izbushka** [comment]
rock.new_btcoex with ant_sel=2 works for me also
HP Pavilion 15-ak073nw
Thanks

**Comment by CaptainChaos** [comment]
Awesome, finally with rock.new_btcoex and ant_sel=1, my wifi on Lenovo Y50-70 works flawlessly. I had issues for months, b/g networks were working, but n networks were unusable. Thanks for your effort, great work!

**Comment by adarshaj** [comment]
HP Pavillion 15 ab522 - works with @lwfinger 's instructions on Ubuntu 14.04!

Thanks! :beers:

**Comment by denevell** [comment]
I've got this working now. Thanks!

Firstly, my wifi card was being blocked by acer-wmi. I did ``modprobe -r acer-wmi`` then suddenly it was not blocked. `rfkill` didn't seem to affect the soft block.

Then I cloned the latest rock.new_btcoex branch. Or in my case zipped it up, put it on a usb to use. Then I ran `modprobe -r rtl8723be` and then deleted them in `/lib/firmware/rtl_wifi` to be extra safe. After that `make install` installed them fine.

After that I had to pass the parameters `modprobe rtl8723be ant_sel=2 ips=0 fwlps=0` with the second antenna being selected, and turning of all the power mode stuff, since the driver was being very, very flaky even though it would connect fine with `ant_sel=2`.

Thanks again. WIll this be merged into main and push up to distributions at any point?

**Comment by BasioMeusPuga** [comment]
Thank you so much for your work! My wifi is where I've run my car over that crap USB dongle I had to buy and I'm no longer considering writing a strongly worded letter to Realtek.

I'd also love to know if/when you'll be merging this into main; and eventually the kernel?

**Comment by laugeo** [comment]
Thanks, 
downloading zip from 
https://github.com/lwfinger/rtlwifi_new/tree/rock.new_btcoex
 then  installing as describe earlier in this page ,
 my HP 250 G4 (xubuntu 15.10) has now working wifi

**Comment by lwfinger** [comment]
What you did is OK, but in general, you should use git to clone the repo rather than downloading a zip. The reason is that if I make any small changes in the code, you will need to download the entire zip. With a git repo, only the new stuff is downloaded using the 'git pull' command.

**Comment by bartbkr** [comment]
Is this fix going to get into the kernel anytime soon? Or at least the master branch of this repo? This is the only fix that makes my wifi card useable.

Thanks for the great work by the way.

**Comment by lwfinger** [comment]
Have you looked at the master branch? The option you want was added March 15!!!!! The necessary revisions were added to the wireless-drivers-next repo last week, and will be in kernel 4.7. Is that soon enough for you? If not, there is nothing I can do. Patches to the kernel move slowly unless they fix a bug. This one is merely a work around for a vendors mistake/incompetence.

**Comment by bartbkr** [comment]
Yes thank you, just tried with master and no issue. Sorry for the confusion.

**Comment by vincentvikram** [comment]
trying to compile the code to address the problem gives me this:

guest@planethacker:~/git/rtlwifi_new$ git checkout rock.new_btcoex 
Branch rock.new_btcoex set up to track remote branch rock.new_btcoex from origin.
Switched to a new branch 'rock.new_btcoex'
guest@planethacker:~/git/rtlwifi_new$ make
make -C /lib/modules/4.6.0-1-amd64/build M=/home/guest/git/rtlwifi_new modules
make[1]: Entering directory '/usr/src/linux-headers-4.6.0-1-amd64'
  CC [M]  /home/guest/git/rtlwifi_new/base.o
  CC [M]  /home/guest/git/rtlwifi_new/cam.o
  CC [M]  /home/guest/git/rtlwifi_new/core.o
  CC [M]  /home/guest/git/rtlwifi_new/debug.o
  CC [M]  /home/guest/git/rtlwifi_new/efuse.o
/home/guest/git/rtlwifi_new/efuse.c:39:31: error: ‘RTL8712_SDIO_EFUSE_TABLE’ defined but not used [-Werror=unused-const-variable=]
 static const struct efuse_map RTL8712_SDIO_EFUSE_TABLE[] = {
                               ^~~~~~~~~~~~~~~~~~~~~~~~
/home/guest/git/rtlwifi_new/efuse.c:35:17: error: ‘MAX_PGPKT_SIZE’ defined but not used [-Werror=unused-const-variable=]
 static const u8 MAX_PGPKT_SIZE = 9;
                 ^~~~~~~~~~~~~~
cc1: all warnings being treated as errors
/usr/src/linux-headers-4.6.0-1-common/scripts/Makefile.build:296: recipe for target '/home/guest/git/rtlwifi_new/efuse.o' failed
make[4]: *** [/home/guest/git/rtlwifi_new/efuse.o] Error 1
/usr/src/linux-headers-4.6.0-1-common/Makefile:1446: recipe for target '_module_/home/guest/git/rtlwifi_new' failed
make[3]: *** [_module_/home/guest/git/rtlwifi_new] Error 2
Makefile:146: recipe for target 'sub-make' failed
make[2]: *** [sub-make] Error 2
Makefile:8: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/usr/src/linux-headers-4.6.0-1-amd64'
Makefile:58: recipe for target 'all' failed
make: *** [all] Error 2

**Comment by lwfinger** [comment]
Setting your compiler to error on ALL warnings is asinine. Change that, or modify the source yourself. As my compiler does not yield any warnings, I will be unable to fix your problems.

In addition, with kernel 4.6, the kernel version is the same as in this repo. Use it.

**Comment by ngergo6** [comment]
Just a heads up for anyone who might have the same environment as me (HP ProBook 455 G2). I managed to solve the problem by installing the drivers from the ```rock.new_btcoex ``` branch and creating a ```/etc/modprobe.d/50-rtl8723be.conf``` file with the following content:

```options rtl8723be ant_sel=1 msi=1 isp=0```

**Comment by jacp01** [comment]
Thank you so much. At last the strange wifi drops on my HP laptop seems to have an explanation, the airplane funtion key (**F12**). In this case, the performance improves with the option ant_sel=1.

**Comment by scribnermj** [comment]
I see quite a lot of content here that reflects the exact issue that I am having. I'm running debian 8.6 with cinnamon. 

dmesg reports 
[19464.310045] rtl8723be: unknown parameter 'ant_sel' ignored
[19464.325857] rtl8723be: Using firmware rtlwifi/rtl8723befw.bin
[19464.325907] rtl8723be 0000:03:00.0: irq 84 for MSI/MSI-X

my rtl8723be.conf looks like this: 
options rtl8723be fwlps=N ips=N ant_sel=2 msi=1

kernal information: 
3.16.0-4-amd64

I have the typical symptoms listed here, no listing of the available networks. 
weak signal when it does connect. 
sometimes is connected only flashes the list when trying to reconnect. 

From what I've read, I may need to upgrade the kernal? Have never done, would that eliminate the need to download a git project and create the new rtl8723be? 

Thanks in advance for your help!

This is an HP laptop, which seems to be one of the more common culprits here.

**Comment by lwfinger** [comment]
This repo is supplied so that you do not need to upgrade the kernel. You build and install this code instead. If you would rather not build an out-of-kernel driver, then you need kernel 4.9. Your distro may have it available.

**Comment by scribnermj** [comment]
I read through quite a lot of the user experiences here and I though it was stated that the kernel should be updated, but it did not help in my case. It was your work here that made the difference. I couldn't believe how easy it and quick it was to get things working.  Thank you for your efforts.

**Comment by piushbansal** [comment]
I don't think the they have really put it in kernel yet. I have kernel 4.10 ,and nothing seems to work. But after using your method , it works though. But I have one problem (and Rajat Goyal asked it earlier but didn't get solved)

> Each time i suspend my laptop and then get it back on . the wifi appears as disabled by hardware. There is no hardware switch and i am not able to use anything to get it back on. I have to reboot my system each time to work around this problem.

except that  in my case it turns airplane mode on, and I just have to manually close it(no restart).
my output of lsmod | grep wmi is:
`hp_wmi                 16384  0
sparse_keymap          16384  1 hp_wmi
mxm_wmi                16384  1 nouveau
wmi                    16384  3 mxm_wmi,nouveau,hp_wmi
rfkill                 20480  7 bluetooth,hp_wmi,cfg80211
`
